cfme/tests/automate/test_common_methods.py

```diff
@@ -320,3 +322,71 @@ def _finalize():
         assert provision_request.is_succeeded(
             method="ui"
         ), f"Provisioning failed: {provision_request.row.last_message.text}"
+
+
+@pytest.fixture(scope="function")
+def vm_folder(provider):
+    """Create Vm folder on VMWare provider"""
+    folder = provider.mgmt.create_folder(fauxfactory.gen_alphanumeric(
+        start="test_folder_", length=20)
+    )
+    yield folder
+    fd = folder.Destroy()
+    wait_for(lambda: fd.info.state == 'success', delay=10, timeout=150)
+
+
+@pytest.mark.tier(3)
+@pytest.mark.ignore_stream("5.10")
+@pytest.mark.provider([VMwareProvider], selector=ONE)
+@pytest.mark.meta(automates=[1716858])
+def test_move_vm_into_folder(appliance, vm_folder, testing_vm, custom_instance):
+    """
+     Bugzilla:
+         1716858
+
+    Polarion:
+        assignee: ghubale
+        casecomponent: Automate
+        initialEstimate: 1/4h
+        tags: automate
+    """
+    script = dedent(
+        f"""
+        vm = $evm.vmdb('vm').find_by_name('{testing_vm.name}')
+        folder = $evm.vmdb('EmsFolder').find_by(:name => '{vm_folder.name}')
+        vm.move_into_folder(folder) unless folder.nil?
+        """
+    )
+    instance = custom_instance(ruby_code=script)
+
+    view = navigate_to(testing_vm, "Details")
+    tree_path = view.sidebar.vmstemplates.tree.currently_selected
+
+    simulate(
+        appliance=appliance,
+        attributes_values={
+            "namespace": instance.klass.namespace.name,
+            "class": instance.klass.name,
+            "instance": instance.name,
+        },
+        message="create",
+        request="Call_Instance",
+        execute_methods=True,
+    )
+    # manipulate tree path. Remove folder - 'Templates' and append with vm_folder name
+    tree_path.pop()
+    tree_path.append(vm_folder.name)
+
+    # Navigating to Vms details page and checking folder of the Vm in accordion of CFME UI
+    view = navigate_to(testing_vm, "Details")
+
+    # Checking new folder appeared
+    def _check():
+        try:
+            view.sidebar.vmstemplates.tree.fill(tree_path)
+            return True
+        except CandidateNotFound:
+            return False
+
+    wait_for(lambda: _check, fail_func=view.browser.refresh, timeout=600, delay=5,
```

Comment:
I'd advice wrapping this into try/except block with pytest.fail.  currently it will appear broken if this wait_for fail.